gpkit/nomials/nomial_math.py

```diff
@@ -699,8 +705,9 @@ def as_posyslt1(self, substitutions=None):
                                   "`.localsolve` instead of `.solve` to"
                                   " form your Model as a SignomialProgram")
 
-    def as_gpconstr(self, x0):
+    def as_gpconstr(self, x0, substitutions=None):
         "Returns GP approximation of an SP constraint at x0"
+        # todo deal with substitutions
```

Comment:
Substitutions get applied at the `as_posyslt1` level; `as_gpconstr` just uses them to check if the constraint will be changing type.